src/internal/connector/sharepoint/data_collections_test.go

```diff
@@ -0,0 +1,313 @@
+package sharepoint_test
+
+import (
+	"context"
+	"testing"
+
+	msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/connector/onedrive"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/pkg/selectors"
+)
+
+// ---------------------------------------------------------------------------
+// consts, mocks
+// ---------------------------------------------------------------------------
+
+const (
+	testBaseDrivePath = "drive/driveID1/root:"
+)
+
+type testFolderMatcher struct {
+	scope selectors.SharePointScope
+}
+
+func (fm testFolderMatcher) IsAny() bool {
+	return fm.scope.IsAny(selectors.SharePointFolder)
+}
+
+func (fm testFolderMatcher) Matches(path string) bool {
+	return fm.scope.Matches(selectors.SharePointFolder, path)
+}
+
+type MockGraphService struct{}
+
+func (ms *MockGraphService) Client() *msgraphsdk.GraphServiceClient {
+	return nil
+}
+
+func (ms *MockGraphService) Adapter() *msgraphsdk.GraphRequestAdapter {
+	return nil
+}
+
+func (ms *MockGraphService) ErrPolicy() bool {
+	return false
+}
+
+// ---------------------------------------------------------------------------
+// tests
+// ---------------------------------------------------------------------------
+
+type SharePointLibrariesIntegrationSuite struct {
+	suite.Suite
+	ctx context.Context
+}
+
+func TestSharePointLibrariesIntegrationSuite(t *testing.T) {
+	if err := tester.RunOnAny(
+		tester.CorsoCITests,
+		tester.CorsoGraphConnectorTests,
+		tester.CorsoGraphConnectorSharePointTests,
+	); err != nil {
+		t.Skip(err)
+	}
+
+	suite.Run(t, new(SharePointLibrariesIntegrationSuite))
+}
+
+func (suite *SharePointLibrariesIntegrationSuite) SetupSuite() {
+	_, err := tester.GetRequiredEnvSls(
+		tester.AWSStorageCredEnvs,
+		tester.M365AcctCredEnvs,
+	)
+
+	require.NoError(suite.T(), err)
+}
+
+func (suite *SharePointLibrariesIntegrationSuite) TestUpdateCollections() {
```

Comment:
Should remove this test completely (I saw the discussion in the previous PR). This is all tested under onedrive already. Recommend we just stick a unit test for the sharepoint matcher in this test and add a unit test for GetCanonicalPath in drive (with a SharepointSource). Other coverage should be through an integration test here for  `sharepoint.DataCollections` that we add here.